src/main/kotlin/org/rust/ide/inspections/fixes/AttachFileToModuleFix.kt

```diff
@@ -0,0 +1,160 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.fixes
+
+import com.intellij.codeInspection.LocalQuickFixOnPsiElement
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.ComboBox
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapiext.isUnitTestMode
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.ui.SimpleListCellRenderer
+import com.intellij.ui.components.dialog
+import com.intellij.ui.layout.panel
+import com.intellij.util.containers.addIfNotNull
+import org.jetbrains.annotations.TestOnly
+import org.rust.cargo.project.model.cargoProjects
+import org.rust.cargo.project.workspace.CargoWorkspace
+import org.rust.ide.notifications.showBalloon
+import org.rust.lang.core.psi.RsFile
+import org.rust.lang.core.psi.RsPsiFactory
+import org.rust.lang.core.psi.ext.RsMod
+import org.rust.lang.core.psi.rustFile
+import org.rust.openapiext.toPsiFile
+
+/**
+ * Attaches a file to a Rust module.
+ *
+ * Before fix:
+ * foo.rs (not attached)
+ * lib.rs
+ *
+ * After fix:
+ * foo.rs
+ * lib.rs
+ *  mod foo;
+ */
+class AttachFileToModuleFix(file: RsFile,
+                            private val targetModuleName: String? = null) : LocalQuickFixOnPsiElement(file) {
+    override fun getFamilyName(): String = text
+    override fun getText(): String = "Attach file to ${targetModuleName ?: "a module"}"
+
+    override fun invoke(project: Project, file: PsiFile, startElement: PsiElement, endElement: PsiElement) {
+        val rsFile = startElement as? RsFile ?: return
+        val availableModules = findAvailableModulesForFile(project, rsFile)
+        if (availableModules.isEmpty()) return
+
+        if (availableModules.size == 1) {
+            insertFileToModule(rsFile, availableModules[0])
+        } else if (availableModules.size > 1) {
+            selectModule(rsFile, availableModules)?.let { insertFileToModule(rsFile, it) }
+        }
+    }
+
+    override fun startInWriteAction(): Boolean {
+        return false
+    }
+
+    companion object {
+        fun findAvailableModulesForFile(project: Project, file: RsFile): List<RsMod> {
+            val virtualFile = file.virtualFile ?: return emptyList()
+            val pkg = project.cargoProjects.findPackageForFile(virtualFile) ?: return emptyList()
+
+            val directory = virtualFile.parent ?: return emptyList()
+            val modules = mutableListOf<RsMod>()
+
+            // mod.rs in the same directory
+            modules.addIfNotNull(findModule(file, project, directory.findFileByRelativePath("mod.rs")))
```

Comment:
Btw all these heuristics are not applicable if the file is mod.rs itself.

Reply:
Good catch, I'll add support for that.